corehq/apps/enterprise/tests/test_decorators.py

```diff
@@ -0,0 +1,64 @@
+from unittest import mock
+
+from django.http import Http404
+from django.test import TestCase, RequestFactory
+
+from corehq.apps.accounting.models import BillingAccount
+from corehq.apps.accounting.tests import generator
+from corehq.apps.domain.models import Domain
+from corehq.apps.enterprise.decorators import require_enterprise_admin
+from corehq.apps.users.models import WebUser
+
+
+class TestRequireEnterpriseAdminDecorator(TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.domain = Domain(
+            name="test-enterprise-001",
+            is_active=True
+        )
+        cls.domain.save()
```

Comment:
This is a pretty common pattern in quite a lot of our tests, so I don't think that's ever been a problem in practice if that is the case. I can definitely see the concern...but I vaguely remember we do some rudimentary cleanup of things like `Domains` in the test couch db before restarting tests. but maybe I'm wrong...? With that in mind, if it's been working for so many other tests, I don't see why it would suddenly fail here.

Reply:
As an anecdotal counterexample, I've already fallen victim to this multiple times. I've had one change in behavior cause setUpClass() to fail, which then repeatedly failed afterwards, since the created object was never deleted. I also personally sometimes just terminate a debugging session, in which case, the tearDown method never runs.  I believe this was the main reason you see a lot of our generator functions generating random names -- since we can't guarantee  couch objects get deleted, we try to use unique identifiers which will never cause a collision. This comes with its own issues, however, and I think the cleanest solution is to just handle situations where your object already exists.